android/src/toga_android/fonts.py

```diff
@@ -21,31 +22,25 @@ class Font:
     def __init__(self, interface):
         self.interface = interface
 
-    def get_size(self):
-        # Default system font size on Android is 14sp (sp = dp, but is separately
-        # scalable in user settings). For what it's worth, Toga's default is 12pt.
-        if self.interface.size == SYSTEM_DEFAULT_FONT_SIZE:
-            font_size = 14
-        else:
-            font_size = self.interface.size
-        return float(font_size)
+    def apply(self, tv, default_size, default_typeface):
+        """Apply the font to the given native widget.
 
-    def get_style(self):
-        if self.interface.weight == BOLD:
-            if self.interface.style == ITALIC:
-                return Typeface.BOLD_ITALIC
-            else:
-                return Typeface.BOLD
-        elif self.interface.style == ITALIC:
-            return Typeface.ITALIC
+        :param tv: A native instance of TextView, or one of its subclasses.
+        """
+        if self.interface.size == SYSTEM_DEFAULT_FONT_SIZE:
+            tv.setTextSize(TypedValue.COMPLEX_UNIT_PX, default_size)
         else:
-            return Typeface.NORMAL
+            # The default size for most widgets is 14sp, so mapping 1 Toga "point" to 1sp
+            # will give relative sizes that are consistent with desktop platforms. It also
+            # means font sizes will all change proportionately if the user adjusts the
+            # text size in the system settings.
+            tv.setTextSize(TypedValue.COMPLEX_UNIT_SP, self.interface.size)
```

Comment:
This is definitely a good approximation; I only wonder if it might be worth introducing a fudge factor so that "12pt == 14sp" (i.e., "size in toga points" * 1.1666 == size in dp).

Reply:
(for the record - this is on the "strong opinion, weakly held" list; I won't give much pushback on a different fudge factor, or no fudge factor at all).

Reply:
Cocoa defaults to 12pt, Windows to 8, and iOS 17, so I don't see any strong reason to fix 12 as the default. Android is already within the range of the other platforms, and making 1 Toga pt = 1sp brings it closer to iOS, which is probably the one it's most useful to be consistent with.

In the long run I think the best solution to this issue is the CSS font size keywords, as mentioned above.

Reply:
Sure - that makes sense.